account_invoice_start_end_dates/models/account_move.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019-2021 Akretion France (https://akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _post(self, soft=True):
+        for move in self:
+            for line in move.line_ids:
+                if (
+                    line.product_id
+                    and line.product_id.must_have_dates
+                    and (not line.start_date or not line.end_date)
+                ):
+                    raise UserError(
+                        _(
```

Comment:
could you apply this change to _ and % similar to this PR?
https://github.com/odoo/odoo/pull/52155